File: clase_estudiante.py
class Estudiante:
    def __init__(self, nombre, edad, promedio):
        self.nombre = nombre
        self.edad = edad
        self.promedio = promedio

    def mostrarInfo(self):
        print(f"Nombre: {self.nombre}, Edad: {self.edad}, Promedio: {self.promedio}")

    def setPromedio(self, nuevo_promedio):
        self.promedio = nuevo_promedio

# 2 Inicialización
e1 = Estudiante("Moisés", 25, 4.0)
e2 = Estudiante("Rafael", 22, 4.5)
e3 = Estudiante("Camila", 21, 3.8)
estudiantes = [e1, e2, e3]

# 3 Recorrido
print("--- Lista de estudiantes original ---")
for e in estudiantes:
    e.mostrarInfo()

# 4 Modificación
e2.setPromedio(4.1)

print("\n--- Lista de estudiantes después de la modificación ---")
for e in estudiantes:
    e.mostrarInfo()

File: clase_estudiante.kt
// 1 Declaración
class Estudiante(var nombre: String, var edad: Int, var promedio: Double) {

    fun mostrarInfo() {
        println("Nombre: $nombre, Edad: $edad, Promedio: $promedio")
    }

}

fun main() {
    // 2 Inicialización
    val e1 = Estudiante("Moisés", 25, 4.0)
    val e2 = Estudiante("Rafael", 22, 4.5)
    val e3 = Estudiante("Camila", 21, 3.8)
    val estudiantes = listOf(e1, e2, e3)

    // 3 Recorrido
    println("--- Lista de estudiantes original ---")
    estudiantes.forEach { it.mostrarInfo() }

    // 4 Modificación
    e2.promedio = 5.0

    println("\n--- Lista de estudiantes después de la modificación ---")
    estudiantes.forEach { it.mostrarInfo() }
}